### PR TITLE
Tweak: Added Grid widget into the widgets panel [ED-11372]

### PR DESCRIPTION
**Pull request by louiswol94**

### CODE REVIEW

**Review by davseve — CHANGES_REQUESTED** [review]
Two general issues:
1. 
image

The grid icon doesn't look like it is taken from the same library as the other icons.
I know it is a design issue, but I am mentioning it.

2. The icon doesn't change on the navigator when I change the container type.

File: includes/base/element-base.php [hunk 1]
```diff
@@ -320,6 +320,21 @@ public function get_default_args( $item = null ) {
...
 	}
 
+	/**
+	 * Get panel presets.
+	 *
+	 * Used for displaying the widget in the panel multiple times, but with different defaults values,
+	 * icon, title etc.
+	 *
+	 * @since 3.16.0
+	 * @access protected
```
Comment (louiswol94) [inline includes/base/element-base.php:330]:
```suggestion
	 * @access public
```

[the remark this answers is not in the document]
Comment (louiswol94) [reply includes/base/element-base.php]: @davseve Done

**Comment by louiswol94** [comment]
> Two general issues: 1. image
> 
> The grid icon doesn't look like it is taken from the same library as the other icons. I know it is a design issue, but I am mentioning it.
> 
> 2. The icon doesn't change on the navigator when I change the container type.

Doron confirmed that it is working correctly.

**Review by davseve — DISMISSED** [review]

File: assets/dev/js/editor/regions/panel/pages/elements/elements.js [hunk 2]
```diff
@@ -160,6 +172,28 @@ PanelElementsLayoutView = Marionette.LayoutView.extend( {
 		this.categoriesCollection = categoriesCollection;
 	},
 
+	shouldNotAddWidget( widget ) {
```
Comment (TzviRabinovitch) [inline assets/dev/js/editor/regions/panel/pages/elements/elements.js:175]: Can this be changed to a positive action?

File: assets/dev/js/editor/regions/panel/pages/elements/elements.js [hunk 3]
```diff
@@ -160,6 +172,28 @@ PanelElementsLayoutView = Marionette.LayoutView.extend( {
...
+		const isContainerActive = elementorCommon.config.experimentalFeatures.container;
+
+		return ! widget.show_in_panel || ( 'inner-section' === widget.name && isContainerActive );
+	},
+
+	deepMerge( originalObj, replacementObj ) {
+		const mergedObj = { ...originalObj };
+
+		for ( const key in replacementObj ) {
+			if ( replacementObj.hasOwnProperty( key ) ) {
```
Comment (TzviRabinovitch) [inline assets/dev/js/editor/regions/panel/pages/elements/elements.js:185]: Can you reduce the nesting? Extract to smaller functions?

		for ( const key in replacementObj ) {
		  deepMergeKey( ...... );
		}

in  deepMergeKey 
if (replacementObj.hasOwnProperty( key){
return 
}

File: assets/dev/js/editor/regions/panel/pages/elements/elements.js (hunk 2 above)
Comment (TzviRabinovitch) [inline assets/dev/js/editor/regions/panel/pages/elements/elements.js:175]: can this function check for positive ( having not in a function name is less readable )

File: assets/dev/js/editor/regions/panel/pages/elements/elements.js [hunk 4]
```diff
@@ -160,6 +172,28 @@ PanelElementsLayoutView = Marionette.LayoutView.extend( {
...
+				if ( 'object' === typeof replacementObj[ key ] && replacementObj[ key ] !== null && originalObj.hasOwnProperty( key ) && 'object' === typeof originalObj[ key ] && originalObj[ key ] !== null ) {
```
Comment (TzviRabinovitch) [inline assets/dev/js/editor/regions/panel/pages/elements/elements.js:186]: can this long condition be extracted to a variable that says what we are checking?

File: assets/dev/js/editor/regions/panel/pages/elements/elements.js (hunk 2 above)
Comment (louiswol94) [reply assets/dev/js/editor/regions/panel/pages/elements/elements.js]: Done

File: assets/dev/js/editor/regions/panel/pages/elements/elements.js (hunk 4 above)
Comment (louiswol94) [reply assets/dev/js/editor/regions/panel/pages/elements/elements.js]: Done

**Review by TzviRabinovitch — APPROVED** [review]

**Review by davseve — APPROVED** [review]

**Pull request merged**